### PR TITLE
Add batch mode to MEBF, MVF, MCR and MDT

### PR DESCRIPTION
**Pull request by S4mpsa**
Same as PAs: Toggle with Sneak+Wire Cutter on the controller. https://github.com/GTNewHorizons/GT5-Unofficial/pull/1446

Increases parallel by up to 128 times (32 768 maximum), while making recipes take up to 128 times longer at the same ratio. This also allows circumventing the 1-tick limit of machines, but is primarily meant for lag reduction.

### CODE REVIEW

**Comment by Connor-Colenso** [comment]
I mean, this is in essence a huge buff for these machines, no?

**Comment by S4mpsa** [comment]
If you consider the 1-tick limit a balance measure, then sure.

It's mostly a buff to MCRs doing low level recipes, and you quickly run into input problems with fluid recipes.

I wouldn't call it a "huge" buff, but I can add a clamp so the processing time doesn't go below 128 ticks in batch mode.

File: src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaChemicalReactor.java [hunk 1]
```diff
@@ -158,6 +160,35 @@ public boolean supportsSingleRecipeLocking() {
...
+        if (aPlayer.isSneaking()) {
+            mUseMultiparallelMode = !mUseMultiparallelMode;
+            if (mUseMultiparallelMode) {
+                GT_Utility.sendChatToPlayer(aPlayer, "Batch recipes");
+            } else {
+                GT_Utility.sendChatToPlayer(aPlayer, "Don't batch recipes");
+            }
+            return true;
+        }
+        return true;
```
Comment (minecraft7771) [inline src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaChemicalReactor.java]: If you do nothing when not sneaking you should return false or maybe even call super but for now the super wouldn't do anything.

File: src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaVacuumFreezer.java [hunk 2]
```diff
@@ -404,6 +406,35 @@ public Object getClientGUI(int aID, InventoryPlayer aPlayerInventory, IGregTechT
...
+        if (aPlayer.isSneaking()) {
+            mUseMultiparallelMode = !mUseMultiparallelMode;
+            if (mUseMultiparallelMode) {
+                GT_Utility.sendChatToPlayer(aPlayer, "Batch recipes");
+            } else {
+                GT_Utility.sendChatToPlayer(aPlayer, "Don't batch recipes");
+            }
+            return true;
+        }
+        return true;
```
Comment (minecraft7771) [inline src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaVacuumFreezer.java]: same

File: src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaBlastFurnace.java [hunk 3]
```diff
@@ -246,10 +248,20 @@ public void onScrewdriverRightClick(byte aSide, EntityPlayer aPlayer, float aX,
...
     public boolean onWireCutterRightClick(
             byte aSide, byte aWrenchingSide, EntityPlayer aPlayer, float aX, float aY, float aZ) {
-        isBussesSeparate = !isBussesSeparate;
-        GT_Utility.sendChatToPlayer(
-                aPlayer, StatCollector.translateToLocal("GT5U.machines.separatebus") + " " + isBussesSeparate);
-        return true;
+        if (aPlayer.isSneaking()) {
+            mUseMultiparallelMode = !mUseMultiparallelMode;
+            if (mUseMultiparallelMode) {
+                GT_Utility.sendChatToPlayer(aPlayer, "Batch recipes");
```
Comment (minecraft7771) [inline src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/multis/mega/GT_TileEntity_MegaBlastFurnace.java]: I know GT as a whole is very inconsistent with this but would be cool if you'd put the strings into the lang file

**Review by minecraft7771 — DISMISSED** [review]

**Comment by Dream-Master** [comment]
@S4mpsa why you close it ?

**Comment by Dream-Master** [comment]
nevermind found new pr

**Pull request closed without merging**